src/runtime/hexagon/hexagon/hexagon_user_dma.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <algorithm>
+
+#include "hexagon_common.h"
+#include "hexagon_user_dma_descriptors.h"
+#include "hexagon_user_dma_instructions.h"
+#include "hexagon_user_dma_registers.h"
+
+namespace tvm {
+namespace runtime {
+namespace hexagon {
+
+int hexagon_user_dma_1d_sync(void* dst, void* src, uint32_t length) {
+#if defined(__hexagon__)
+  // not thread safe
+  static int config_dma = 0;
```

Comment:
Extract the DMA config code below into a separate function returning int, and change this declaration to `static int config_dma = initialization_function();`  This will make it thread-safe.